client.go

```diff
@@ -35,14 +38,28 @@ type RESTClient struct {
 
 // NewRESTClient constructs a new REST client containing each sub client.
 func NewRESTClient(cl *client.Harbor, authInfo runtime.ClientAuthInfoWriter) *RESTClient {
-	c := &RESTClient{
+	return &RESTClient{
 		user:        user.NewClient(cl, authInfo),
 		project:     project.NewClient(cl, authInfo),
 		registry:    registry.NewClient(cl, authInfo),
 		replication: replication.NewClient(cl, authInfo),
 		system:      system.NewClient(cl, authInfo),
 	}
-	return c
+}
+
+// NewRESTClientWithSwaggerClient constructs a new REST client containing a swagger
+// API client using the defined host string + basePath as well as basic auth info.
+func NewRESTClientWithSwaggerClient(host, basePath, username, password string) *RESTClient {
```

Comment:
IMO, this name breaks abstraction a bit since it exposes quite a lot on the internal workings of this method (a user probably won't care about any swagger clients). Maybe...

```suggestion
func NewRESTClientForHost(host, basePath, username, password string) *RESTClient {
```

❓